apps/prairielearn/src/pages/administratorQueries/administratorQueries.ts

```diff
@@ -0,0 +1,53 @@
+import { z } from 'zod';
+import * as express from 'express';
+import * as asyncHandler from 'express-async-handler';
+import * as fsPromises from 'node:fs/promises';
+import * as path from 'path';
+
+import jsonLoad = require('../../lib/json-load');
+import { AdministratorQueries, type AdministratorQuery } from './administratorQueries.html';
+
+const router = express.Router();
+const queriesDir = path.resolve(__dirname, '..', '..', 'admin_queries');
+
+const AdministratorQueryJsonParamsSchema = z.object({
+  name: z.string(),
+  description: z.string(),
+  default: z.string().nullable().optional(),
+  comment: z.any().nullable(),
+});
+
+const AdministratorQueryJsonSchema = z.object({
+  description: z.string(),
+  resultFormats: z.any().nullable(),
+  comment: z.any().nullable(),
+  params: z.array(AdministratorQueryJsonParamsSchema).nullable().optional(),
+});
+
+router.get(
+  '/',
+  asyncHandler(async (req, res, _next) => {
+    const fileList = await fsPromises.readdir(queriesDir);
+    const jsonList = fileList.filter((f) => /\.json$/.test(f));
+    const queries = await Promise.all(
+      jsonList.map(async (f) => {
+        const contents = AdministratorQueryJsonSchema.parse(
+          await jsonLoad.readJSONAsync(path.join(queriesDir, f)),
+        );
+        return {
+          ...contents,
+          sqlFilename: f.replace(/\.json$/, '.sql'),
+          link: f.replace(/\.json$/, ''),
+        };
+      }),
+    );
+    res.send(
+      AdministratorQueries({
+        resLocals: res.locals,
+        queries: queries as AdministratorQuery[],
```

Comment:
Having to do `as ...` is a bit of a code smell - why was this necessary?

Reply:
Code smell was the result of using `any` types. Added a zod schema in commit 782f18f to address the shape of the queries and remove the use of `any` types.

Reply:
I believe this change now makes the page throw an error on load. I'd recommend always doing a quick manual functionality test before pushing new commits.

Reply:
Resolved in commit 2a7e7b8.